stdlib/database/sql/src/main/java/org/ballerinalang/database/sql/SQLDatasourceUtils.java

```diff
@@ -995,6 +997,13 @@ public static String getSQLType(BType value) {
             return Constants.SQLDataTypes.BOOLEAN;
         case TypeTags.DECIMAL_TAG:
             return Constants.SQLDataTypes.DECIMAL;
+        case TypeTags.ARRAY_TAG:
+            if ((value instanceof BArrayType) && ((BArrayType) value).getElementType().getTag() == TypeTags.BYTE_TAG) {
+                return Constants.SQLDataTypes.BINARY;
+            } else {
+                throw new BallerinaException("Found unsupported array data type as direct value and supported array " +
```

Comment:
```suggestion
                throw new BallerinaException("Array data type as direct value is supported only with byte type elements, use sql:Parameter " +
```

Can we have something like this?